CovertInfiltration/Src/CovertInfiltration/Classes/X2EventListener_Infiltration_UI.uc

```diff
@@ -426,18 +427,49 @@ static protected function EventListenerReturn ShortcutsResistanceButtonVisible(O
 
 static function EventListenerReturn UpdateResources(Object EventData, Object EventSource, XComGameState GameState, Name Event, Object CallbackData)
 {
+	local XComGameState_HeadquartersXcom XComHQ;
 	local UIAvengerHUD AvengerHUD;
 	local UIScreenStack ScreenStack;
-
 	local UIScreen CurrentScreen;
 	local UICovertActionsGeoscape CovertActions;
-
+	local int CurrentBarracksSize, CurrentBarracksLimit, MessageColor;
+	
+	XComHQ = `XCOMHQ;
 	AvengerHUD = `HQPRES.m_kAvengerHUD;
 	ScreenStack = AvengerHUD.Movie.Pres.ScreenStack;
-
 	CurrentScreen = ScreenStack.GetCurrentScreen();
-	CovertActions = UICovertActionsGeoscape(ScreenStack.GetFirstInstanceOf(class'UICovertActionsGeoscape'));
+	
+	if (UIFacility_LivingQuarters(CurrentScreen) != none ||
+		UIStrategyMap(CurrentScreen) != none ||
+		UIFacilityGrid(CurrentScreen) != none ||
+		UIRecruitSoldiers(CurrentScreen) != none ||
+		(UIChooseUpgrade(CurrentScreen) != none && UIFacility_LivingQuarters(ScreenStack.GetFirstInstanceOf(class'UIFacility_LivingQuarters')) != none))
+	{
+		CurrentBarracksSize = XComHQ.GetNumberOfSoldiers() + XComHQ.GetNumberOfScientists() + XComHQ.GetNumberOfEngineers();
+		CurrentBarracksLimit = class'XComGameState_CovertInfiltrationInfo'.static.GetInfo().CurrentBarracksLimit;
 
+		if (CurrentBarracksSize > CurrentBarracksLimit)
+		{
+			MessageColor = eUIState_Bad;
+		}
+		else if (CurrentBarracksSize == CurrentBarracksLimit)
+		{
+			MessageColor = eUIState_Warning;
+		}
+		else
+		{
+			MessageColor = eUIState_Cash;
+		}
+		
+		AvengerHUD.AddResource(default.strBarracksSizeTitle, class'UIUtilities_Text'.static.GetColoredText(CurrentBarracksSize $ "/" $ CurrentBarracksLimit, MessageColor));
+		
+		AvengerHUD.ShowResources();
+
+		return ELR_NoInterrupt;
```

Comment:
I dislike this return. It may break other functionality. Get rid of it.

Also get rid of `if (CovertActions == none) return ELR_NoInterrupt;` below